e2e/cp_test.go

```diff
@@ -4596,3 +4596,160 @@ func TestCopyS3ObjectsWithIncludeExcludeFilter2(t *testing.T) {
+	assert.Assert(t, fs.Equal(workdir.Path(), expected))
+
+	// assert lines
+	assertLines(t, result.Stdout(), map[int]compareFunc{
+		0: equals("cp %v/a/another_test_file.txt %sa/another_test_file.txt", workdir.Path(), dst),
+		1: equals("cp %v/file1.html %sfile1.html", workdir.Path(), dst),
+	}, sortInput(true))
+
+	// assert s3
+	assert.Assert(t, ensureS3Object(s3client, bucket, "file1.html", "<html><body><h1>file1</h1></body></html>", ensureContentType("video/avi")))
+	assert.Assert(t, ensureS3Object(s3client, bucket, "a/another_test_file.txt", "yet another txt file. yatf.", ensureContentType("video/avi")))
+}
+
+// cp --content-type "video/avi" s3://srcbucket/object s3://dstbucket/
+func TestCopySingleS3ObjectToAnotherBucketWithContentType(t *testing.T) {
+	t.Parallel()
+
+	srcbucket := s3BucketFromTestNameWithPrefix(t, "src")
+	dstbucket := s3BucketFromTestNameWithPrefix(t, "dst")
+
+	s3client, s5cmd := setup(t)
+
+	createBucket(t, s3client, srcbucket)
+	createBucket(t, s3client, dstbucket)
+
+	const (
+		filename = "testfile.txt"
+		content  = "this is a test file"
+	)
+
+	putFile(t, s3client, srcbucket, filename, content)
+
+	src := fmt.Sprintf("s3://%v/%v", srcbucket, filename)
+	dst := fmt.Sprintf("s3://%v/", dstbucket)
+
+	cmd := s5cmd("cp", "--content-type", "video/avi", src, dst)
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Success)
+
+	// assert s3
+	assert.Assert(t, ensureS3Object(s3client, srcbucket, filename, content))
+	assert.Assert(t, ensureS3Object(s3client, dstbucket, filename, content, ensureContentType("video/avi")))
+}
+
+// cp --content-type "video/avi" s3://srcbucket/* s3://dstbucket/
+func TestCopyMultipleS3ObjectsToAnotherBucketWithContentType(t *testing.T) {
+	t.Parallel()
+
+	srcbucket := s3BucketFromTestNameWithPrefix(t, "src")
+	dstbucket := s3BucketFromTestNameWithPrefix(t, "dst")
+
+	s3client, s5cmd := setup(t)
+
+	createBucket(t, s3client, srcbucket)
+	createBucket(t, s3client, dstbucket)
+
+	fileAndContent := map[string]string{
+		"file1.txt":               "this is a test file 1",
+		"a/another_test_file.txt": "yet another txt file. yatf.",
+	}
+
+	for filename, content := range fileAndContent {
+		putFile(t, s3client, srcbucket, filename, content)
+	}
+
+	src := fmt.Sprintf("s3://%v/*", srcbucket)
+	dst := fmt.Sprintf("s3://%v/", dstbucket)
+
+	cmd := s5cmd("cp", "--content-type", "video/avi", src, dst)
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Success)
+
+	// assert s3
+	// these seperation is intentional to make it easier to debug.
+	for filename, content := range fileAndContent {
+		assert.Assert(t, ensureS3Object(s3client, srcbucket, filename, content))
+	}
+
+	for filename, content := range fileAndContent {
+		assert.Assert(t, ensureS3Object(s3client, dstbucket, filename, content))
+	}
+
+	for filename, content := range fileAndContent {
+		assert.Assert(t, ensureS3Object(s3client, dstbucket, filename, content, ensureContentType("video/avi")))
+	}
```

Comment:
Why is it required to assert multiple times after no modification ? Is this on purpose ?

Reply:
yes it was but I am gonna delete first two.

Reply:
done.